application/server/server_smoke_test.go

```diff
@@ -605,23 +611,67 @@ func checkForScanParams(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder,
 		}
 		return false
 	}, 10*time.Second, 10*time.Millisecond)
-	return scanParams
 }
 
-func checkAutofixDiffs(t *testing.T, c *config.Config, snykCodeScanParams types.SnykScanParams, loc server.Local) {
+func getIssueListFromPublishDiagnosticsNotification(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder, p product.Product, folderPath string) []types.ScanIssue {
+	t.Helper()
+
+	var issueList []types.ScanIssue
+	notifications := jsonRPCRecorder.FindNotificationsByMethod("textDocument/publishDiagnostics")
+	for _, n := range notifications {
+		diagnosticsParams := types.PublishDiagnosticsParams{}
+		_ = n.UnmarshalParams(&diagnosticsParams)
+		for _, diagnostic := range diagnosticsParams.Diagnostics {
+			diagnosticCode, ok := diagnostic.Code.(string)
+			if ok && diagnosticCode == "Snyk Error" {
+				continue
+			}
+			if diagnostic.Source != string(p) || filepath.Dir(uri.PathFromUri(diagnosticsParams.URI)) != folderPath {
+				continue
+			}
+
+			var issue types.ScanIssue
+			err := mapSerializedIssueToStruct(diagnostic.Data, &issue)
+
+			if err != nil {
+				continue
```

Comment:
Why don't we fail the test here? Shouldn't we be able to map all returned issues?